### DISCUSSION TITLE
Enable grammarly for tex files?

### DISCUSSION THREAD

**Discussion by samerickson** [williamboman/nvim-lsp-installer#421] [Q&A] [answered]
Hello,

I am trying to get Grammarly to work on latex files. I have Grammarly installed and it works with markdown files, but I am unable to get it to work with `.tex` files.

I have tried adding `filetypes = { "tex" }` and `filetypes = { "latex" } to the opts in my Grammarly server setup function, but neither of these seem to work.

I have also tried to start the server using `:LspStart Grammarly` this also does not work.

Has anyone run into this or know how to configure the Grammarly server so that it runs on Latex files?

**Comment by williamboman** [comment, marked as answer]
Hello @samerickson!

When setting up the server - is this done via nvim-lsp-installer APIs, for example like in [this snippet](https://github.com/williamboman/nvim-lsp-installer/wiki/Advanced-Configuration#overriding-the-default-lsp-server-options)?

Also, just making sure, is the `'filetype'` correctly set when you open these files? Does `:echo &ft` produce `tex`?

I know the Grammarly server is a bit unstable and struggle with certain files at the moment. I tried this out myself on `.tex` files, and it seems like it's struggling (see screenshot below). There's some more background in this [PR in lspconfig](https://github.com/neovim/nvim-lspconfig/pull/1562).

<img width="862" alt="Screenshot 2022-01-12 at 22 37 00" src="https://user-images.githubusercontent.com/6705160/149225062-f6943712-cdfe-4cc5-af4d-64aa09465d5e.png">

**Reply by samerickson** [reply]
Hello @williamboman,

Yes, I sometimes have to set the `filetype` manually for some reason. 

Even through I have Grammarly installed (and working on markdown files), when I enter `LspInfo` I get the following:
![image](https://user-images.githubusercontent.com/9330283/149226252-ef762231-4a67-4a88-882a-f96e69c10a35.png)

**Reply by williamboman** [reply]
Hm that seems a bit odd. Are you overriding the `grammarly` filetype triggers? If yes, what if you don't override and leave it to the defaults?

**Reply by samerickson** [reply]
I am not overriding the default filetype triggers. In fact, I have tried overriding them without success.

**Reply by williamboman** [reply]
@samerickson I found your dotfiles and opened a PR - https://github.com/samerickson/dotfiles/pull/1. Ought to fix it! I'm not sure how well the grammarly service works with tex files, so no guarantees there!

**Reply by samerickson** [reply]
@williamboman Thank you so much! You have just saved me so much time on my research project!
